### PR TITLE
[Docs] Add TGATE in section `optimization`

### PR DESCRIPTION
**Pull request by WentianZhang-ML**
# What does this PR do?

This PR is for the documentation of [TGATE](https://arxiv.org/abs/2404.02747), which can accelerate inferences of `PixArtAlphaPipeline`, `StableDiffusionPipeline`, and `StableDiffusionXLPipeline` by skipping the calculation of cross-attention once it converges. The doc includes the instructions for installing and using TGATE, along with a comparison overview with SD-1.5, SD-2.1, SD-XL, PixArt-Alpha, DeepCache, and LCM, both with or without TGATE.

### 🚀 Major Features 

* Training-Free.
* Easily Integrate into [Diffusers](https://github.com/huggingface/diffusers/tree/main).
* Only a few lines of code are required.
* Complementary to [DeepCache](https://github.com/horseee/DeepCache).
* Friendly support [Stable Diffusion pipelines](https://huggingface.co/stabilityai), [PixArt](https://pixart-alpha.github.io/), and [Latent Consistency Models](https://latent-consistency-models.github.io/).
* 10%-50% speed up for different models. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
   [Add an option to use TGATE method to accelerate the inference of PixArt-Alpha. #7596 ](https://github.com/huggingface/diffusers/pull/7596)

- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sayakpaul @yiyixuxu

### CODE REVIEW

File: docs/source/en/optimization/tgate.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+* Friendly support [Stable Diffusion pipelines](https://huggingface.co/stabilityai), [PixArt](https://pixart-alpha.github.io/), and [Latent Consistency Models](https://latent-consistency-models.github.io/).
+* 10%-50% speed up for different models. 
+
+## 📖 Quick Start
+
+### 🛠️ Installation
+
+Start by installing [TGATE](https://github.com/HaozheLiu-ST/T-GATE/tree/release-v.0.1.0):
+
+```
```
Comment (stevhliu) [inline docs/source/en/optimization/tgate.md:25]:
```suggestion
```bash
```

File: docs/source/en/optimization/tgate.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+pip install tgate
+```
```
Comment (stevhliu) [inline docs/source/en/optimization/tgate.md:27]:
```suggestion
pip install tgate
pip install -U pytorch diffusers transformers accelerate DeepCache
```

File: docs/source/en/optimization/tgate.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+* pytorch>=2.0.0
+* diffusers>=0.27.2
+* transformers==4.37.2
+* DeepCache==0.1.1
+* accelerate
+
+### 🌟 Usage
+
+Accelerate `PixArtAlphaPipeline` with TGATE:
```
Comment (stevhliu) [inline docs/source/en/optimization/tgate.md:39]: Feel free to update table with any additional `TgateLoader`'s. Also if you can fill in the parentheses with a brief explanation of what the gate step does that'd be great!

```suggestion
To use T-GATE with a pipeline, you need to use its corresponding loader.

| Pipeline | T-GATE Loader |
|---|---|
| PixArt | TgatePixArtLoader |
| Stable Diffusion XL | TgateSDXLLoader |
| Stable Diffusion XL + DeepCache | TgateSDXLDeepCacheLoader |

Next, create a `TgateLoader` with a pipeline, the gate step(`add brief description here`), and the number of inference steps. Then call the `tgate` method on the pipeline with a prompt, gate step, and the number of inference steps.

Let's see how to enable this for several different pipelines.
```

File: docs/source/en/optimization/tgate.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
++        num_inference_steps=inference_step,
++ )
+pipe = pipe.to("cuda")
+
++ image = pipe.tgate(
++         "An alpaca made of colorful building blocks, cyberpunk.",
++         gate_step=gate_step,
++         num_inference_steps=inference_step,
++ ).images[0]
+```
```
Comment (stevhliu) [inline docs/source/en/optimization/tgate.md:62]: 1. It'd be easier for users to copy/paste code snippet without `diff`  
2. We can also wrap all these pipeline examples with our doc-builders special syntax so users can easily switch between the different code samples without scrolling.

```md
<hfoptions id="pipelines">
<hfoption id="PixArt">

code here

</hfoption>
<hfoption id="Stable Diffusion XL">

code here

</hfoption>
<hfoption id="StableDiffusionXL with DeepCache">

code here

</hfoption>
<hfoption id="Latent Consistency Model">

code here

</hfoption>
</hfoptions>
```

```suggestion
```py
import torch
from diffusers import PixArtAlphaPipeline
from tgate import TgatePixArtLoader

pipe = PixArtAlphaPipeline.from_pretrained("PixArt-alpha/PixArt-XL-2-1024-MS", torch_dtype=torch.float16)
pipe = TgatePixArtLoader(
       pipe,
       gate_step=8,
       num_inference_steps=25,
)
pipe = pipe.to("cuda")

image = pipe.tgate(
       "An alpaca made of colorful building blocks, cyberpunk.",
        gate_step=gate_step,
       num_inference_steps=inference_step,
).images[0]
```
```

File: docs/source/en/optimization/tgate.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
++ image = pipe.tgate(
++         "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k.",
++         gate_step=gate_step,
++         num_inference_steps=inference_step
++ ).images[0]
+```
+
+TGATE also supports `StableDiffusionPipeline` and `PixArt-alpha/PixArt-LCM-XL-2-1024-MS`.
+More details can be found at [here](https://github.com/HaozheLiu-ST/T-GATE/tree/release-v.0.1.0/main.py).
```
Comment (stevhliu) [inline docs/source/en/optimization/tgate.md:170]:
```suggestion
T-GATE also supports [`StableDiffusionPipeline`] and [PixArt-alpha/PixArt-LCM-XL-2-1024-MS](https://hf.co/PixArt-alpha/PixArt-LCM-XL-2-1024-MS).
```

File: docs/source/en/optimization/tgate.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+## 📄 Results
```
Comment (stevhliu) [inline docs/source/en/optimization/tgate.md:172]:
```suggestion
## Benchmarks
```

File: docs/source/en/optimization/tgate.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+| LCM (SD-XL)           | 11.955T  | 2.570B    | 3.805s  | 25.044                    |
+| LCM w/ TGATE          | 11.171T  | 2.024B    | 3.533s  | 25.028                    |
+| LCM (Pixart-Alpha)    | 8.563T   | 611.350M  | 4.733s  | 36.086                    |
+| LCM w/ TGATE          | 7.623T   | 462.585M  | 4.543s  | 37.048                    |
+
+The latency is tested on a 1080ti commercial card. 
+
+The MACs and Params are calculated by [calflops](https://github.com/MrYxJ/calculate-flops.pytorch). 
+
+The FID is calculated by [PytorchFID](https://github.com/mseitzer/pytorch-fid).
```
Comment (stevhliu) [inline docs/source/en/optimization/tgate.md:194]: Move this right after the "Benchmarks"

```suggestion
The latency is tested on an NVIDIA 1080TI, MACs and Params are calculated with [calflops](https://github.com/MrYxJ/calculate-flops.pytorch), and the FID is calculated with [PytorchFID](https://github.com/mseitzer/pytorch-fid).
```

**Comment by WentianZhang-ML** [comment]
Thanks for your review! I have committed suggestions and also made some modifications.  :) @stevhliu

**Review by stevhliu — APPROVED** [review]
LGTM, thanks for adding this! 🤗

**Comment by WentianZhang-ML** [comment]
Hi, @stevhliu. Sorry for taking your time. 

We opened a new PR #7698 to fix bugs in the code snippet of PixArt (lines 46 and 47), where gate_step and num_inference_steps were not assigned.

**Pull request merged**